contrib/win32/win32compat/win32_pty.c

```diff
@@ -90,10 +92,42 @@ CreateConPty(const wchar_t *cmdline,
 	return 0;
 }
 
-int is_conpty_supported()
+int
+is_conpty_supported()
 {
-	/* TODO - put conditional logic in here that determines if conpty is supported */
-	return 0;
+	wchar_t system32_path[PATH_MAX] = { 0, };
+	wchar_t kernelbase_dll_path[PATH_MAX] = { 0, };
+	HMODULE hm_kernelbase = NULL;
+	int retVal = 0;
+
+	if (isConpty != -1)
+		return isConpty;
+
+	if (!GetSystemDirectoryW(system32_path, PATH_MAX)) {
+		error("failed to get system directory");
+		goto done;
+	}
+
+	wcscat_s(kernelbase_dll_path, PATH_MAX, system32_path);
+	wcscat_s(kernelbase_dll_path, PATH_MAX, L"\\KernelBase.dll");
+
+	if ((hm_kernelbase = LoadLibraryW(kernelbase_dll_path)) == NULL) {
+		error("failed to load kernerlbase dll:%s", kernelbase_dll_path);
```

Comment:
When this dll does not exist, does it mean conpty is not supported or it is error?

Reply:
This DLL is supposed to be there from windows7.
Incase if we couldn't find the DLL then we treat as if conpty is not supported.